### PR TITLE
Update endpoint access examples: added info about accessing auth-protected apps

### PR DESCRIPTION
**Pull request by dmitsf**
## What does this PR do?

Adds more detailed examples: now the info on how to access to the basic auth-protected app is added.

Screen Shot 2022-12-20 at 12 15 00 PM

## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- [ ] Did you list all the **breaking changes** introduced by this pull request?
- [ ] Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [ ] Check that all items from **Before submitting** are resolved
- [ ] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

cc @borda

### CODE REVIEW

File: src/lightning_app/components/serve/python_server.py [hunk 1]
```diff
@@ -70,8 +70,15 @@ def request_code_sample(url: str) -> str:
...
-    "image": img
-})"""
+"image": img
+})
+# If you are using basic authentication for your app, you should add your credentials to the request:
+# response = requests.post('"""
+            + url
+            + """', json={
+# "image": img
+# }, auth=requests.auth.HTTPBasicAuth('your_username', 'your_password'))
```
Comment (awaelchli) [inline src/lightning_app/components/serve/python_server.py:80]: the comment marker for these lines is probably not intentional?

File: src/lightning_app/components/serve/python_server.py [hunk 2]
```diff
@@ -112,8 +119,14 @@ def request_code_sample(url: str) -> str:
 response = requests.post('"""
             + url
             + """', json={
-    "text": "A portrait of a person looking away from the camera"
+"text": "A portrait of a person looking away from the camera"
 })
+# If you are using basic authentication for your app, you should add your credentials to the request:
+# response = requests.post('"""
```
Comment (awaelchli) [inline src/lightning_app/components/serve/python_server.py:125]: here too

**Review by awaelchli — APPROVED** [review]

**Review by Borda — APPROVED** [review]

File: src/lightning_app/components/serve/python_server.py (hunk 1 above)
Comment (dmitsf) [reply src/lightning_app/components/serve/python_server.py:80]: It's intentional - just to (1) keep both options (creds-less and with credentials) at the same example and (2) keep the previous option (without credentials) as a default one.

File: src/lightning_app/components/serve/python_server.py (hunk 2 above)
Comment (dmitsf) [reply src/lightning_app/components/serve/python_server.py:125]: (the same - this is intentional)

File: src/lightning_app/components/serve/python_server.py (hunk 1 above)
Comment (awaelchli) [reply src/lightning_app/components/serve/python_server.py:80]: That's very unfortunate. If the user copies the code here (which is I assume why we show it), they get a syntax error due to missing braces. 
Also, the code formatting would look much more readable if we used `textwrap.dedent`.

**Pull request merged**